helper/reader.go

```diff
@@ -0,0 +1,51 @@
+package helper
+
+import (
+	"io"
+)
+
+type MultiDimByteReader struct {
+	s                          [][]byte
+	currentSliceIndex          int
```

Comment:
What does `current` mean? We can also add comment statements on the top of related fields. This comments should be include what and why maybe.